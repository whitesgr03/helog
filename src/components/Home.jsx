import style from "../styles/Home.module.css";

import { Link } from "react-router-dom";

const Home = () => (
	<div className={style.home}>
		<h2 className={style.title}>A Daily Story for Reader.</h2>
		<p>
			Lorem ipsum dolor sit amet consectetur, adipisicing elit. Odio,
			veniam voluptatem veritatis maxime iste rerum quo blanditiis cum ea
			nulla officiis neque molestiae voluptate qui adipisci velit magnam
			consequatur. Quo. Quia voluptas consequuntur assumenda ut totam
			quaerat explicabo eum nemo? Eveniet iste neque eligendi eius
			provident doloremque magnam maiores culpa repudiandae ipsam.
			Repudiandae illum facilis, omnis qui expedita debitis ea.
		</p>
		<Link to={`/`} className={style.link}>
			Latest Posts
			<span className={`icon ${style.doubleArrowRight}`} />
		</Link>
	</div>
);

export default Home;
