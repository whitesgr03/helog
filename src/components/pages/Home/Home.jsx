// Packages
import { Link } from 'react-router-dom';

// Styles
import styles from './Home.module.css';
import imageStyles from '../../../styles/image.module.css';

// Components
import { Posts } from '../Post/Posts';

// Variables
import url from '../../../assets/hero.jpg';

export const Home = () => {
	return (
		<div className={styles.home}>
			<div className={styles.container}>
				<div className={styles.wrap}>
					<div className={imageStyles.content}>
						<img src={url} />
					</div>
				</div>
				<div className={styles.content}>
					<h2 className={styles.title}>A Daily Stories For Reader</h2>
					<p>
						The personal blog is an ongoing online diary or commentary written
						by an individual, rather than a corporation or organization. While
						the vast majority of personal blogs attract very few readers, other
						than the blogger&apos;s immediate family and friends, a small number
						of personal blogs have become popular, to the point that they have
						attracted lucrative advertising sponsorship. A tiny number of
						personal bloggers have become famous, both in the online community
						and in the real world.
					</p>
					<Link to={'/posts'} className={styles.link}>
						<span>Latest </span>
						<span>All </span>Posts
						<span
							className={`${imageStyles.icon} ${styles.doubleArrowRight}`}
						/>
					</Link>
				</div>
			</div>
			<div className={styles['latest-posts']}>
				<h2>Latest Posts</h2>
				<Posts limit={4} />
			</div>
		</div>
	);
};
