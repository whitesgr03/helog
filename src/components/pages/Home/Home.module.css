.home {
	padding: 5%;
}

.container {
	height: 100%;
}

.wrap {
	display: none;
}

.content {
	display: flex;
	row-gap: 30px;
	flex-direction: column;
	justify-content: space-evenly;
	height: 100%;
}

.title {
	color: var(--light-scheme-highlight);
	font-size: 2.5rem;
}

.link {
	display: flex;
	column-gap: 5px;
	align-items: center;
	align-self: flex-start;
	border: 2px solid var(--light-scheme-color);
	border-radius: 20px;
	padding: 10px 20px;
}

.link span:nth-child(2) {
	display: none;
}

.latest-posts {
	display: none;
}

.double-arrow-right {
	mask-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 15 15'%3E%3Cpath fill='%23000' fill-rule='evenodd' d='M2.146 11.146a.5.5 0 0 0 .708.708l4-4a.5.5 0 0 0 0-.708l-4-4a.5.5 0 1 0-.708.708L5.793 7.5zm6 0a.5.5 0 0 0 .708.708l4-4a.5.5 0 0 0 0-.708l-4-4a.5.5 0 1 0-.708.708L11.793 7.5z' clip-rule='evenodd'/%3E%3C/svg%3E");
	width: var(--icon-size-desktop);
	height: var(--icon-size-desktop);
}

/* Pseudo classes */
.link:hover {
	background-color: var(--light-scheme-color);
	color: var(--dark-scheme-color);
}

/* Dark scheme classes */
:global(.dark) .content {
	color: var(--dark-scheme-color);
}

:global(.dark) .title {
	color: var(--dark-scheme-highlight);
}

:global(.dark) .link {
	border-color: var(--dark-scheme-color);
	color: var(--dark-scheme-color);
}

:global(.dark) .link:hover {
	background-color: var(--dark-scheme-color);
	color: var(--light-scheme-color);
}

@media (min-width: 450px) {
	.home {
		padding: 5% 10%;
	}
	.link {
		align-self: flex-end;
	}
}

@media (min-width: 700px) {
	.home {
		padding: 5% 15%;
	}

	.container {
		margin-bottom: 5%;
		height: auto;
	}
	.content {
		row-gap: 40px;
		height: auto;
	}

	.link span:nth-child(1) {
		display: none;
	}

	.link span:nth-child(2) {
		display: inline;
	}

	.latest-posts {
		display: flex;
		row-gap: 20px;
		flex-direction: column;
	}

	.latest-posts ul {
		display: grid;
		grid-template-columns: 1fr 1fr;
		grid-auto-rows: 1fr;
		gap: 30px;
	}

	.latest-posts ul li {
		padding: 0;
	}

	/* Dark scheme classes */
	:global(.dark) .latest-posts {
		color: var(--dark-scheme-color);
	}
}

@media (min-width: 1000px) {
	.home {
		padding: 0;
	}
	.container {
		display: flex;
		margin-bottom: 0;
	}

	.wrap {
		display: block;
		display: flex;
		flex: 1 1 70%;
		align-items: stretch;
	}

	.content {
		flex: 1 1 30%;
		border-bottom: 2px solid var(--light-scheme-color);
		padding: 20px;
	}

	.latest-posts {
		padding: 5% 10%;
	}

	.latest-posts ul {
		grid-template-columns: 1fr 1fr 1fr 1fr;
	}

	.latest-posts ul li {
		overflow: hidden;
	}

	/* Dark scheme classes */
	:global(.dark) .content {
		border-color: var(--dark-scheme-color);
	}
}

@media (min-width: 1300px) {
	.home {
		padding: 5%;
	}
	.container {
		display: block;
		position: relative;
	}
	.content {
		position: absolute;
		top: 0;
		left: 0;
		backdrop-filter: blur(4px);
		border-bottom: 0;
		background-color: rgba(199, 202, 207, 0.4);
		padding: 0 20%;
		width: 100%;
		height: 100%;
		line-height: 2rem;
	}

	.content h2 {
		text-shadow:
			var(--dark-scheme-color) 0 0 10px,
			var(--dark-scheme-color) 0 0 10px,
			var(--dark-scheme-color) 0 0 10px,
			var(--dark-scheme-color) 0 0 10px,
			var(--dark-scheme-color) 0 0 10px;
	}

	.content p {
		color: var(--light-scheme-color);
		font-size: 1.2rem;
		line-height: 3rem;
		text-shadow:
			var(--dark-scheme-color) 0 0 10px,
			var(--dark-scheme-color) 0 0 10px,
			var(--dark-scheme-color) 0 0 10px,
			var(--dark-scheme-color) 0 0 10px,
			var(--dark-scheme-color) 0 0 10px;
	}

	.link {
		border: 2px solid transparent;
		background-color: var(--light-scheme-color);
		color: var(--dark-scheme-color);
	}

	.latest-posts {
		padding: 0;
		padding-top: 5%;
	}

	/* Pseudo classes */
	.link:hover {
		border: 2px solid var(--light-scheme-color);
		background-color: transparent;
		color: var(--light-scheme-color);
	}

	/* Dark scheme classes */
	:global(.dark) .content {
		background-color: rgba(0, 0, 0, 0.4);
	}

	:global(.dark) .content h2 {
		text-shadow:
			var(--light-scheme-color) 0 0 10px,
			var(--light-scheme-color) 0 0 10px,
			var(--light-scheme-color) 0 0 10px,
			var(--light-scheme-color) 0 0 10px,
			var(--light-scheme-color) 0 0 10px;
	}

	:global(.dark) .content p {
		color: var(--dark-scheme-color);
		text-shadow:
			var(--light-scheme-color) 0 0 10px,
			var(--light-scheme-color) 0 0 10px,
			var(--light-scheme-color) 0 0 10px,
			var(--light-scheme-color) 0 0 10px,
			var(--light-scheme-color) 0 0 10px;
	}

	:global(.dark) .link {
		background-color: var(--dark-scheme-color);
		color: var(--light-scheme-color);
	}

	:global(.dark) .link:hover {
		border: 2px solid var(--dark-scheme-color);
		background-color: transparent;
		color: var(--dark-scheme-color);
	}
}

@media (min-width: 1600px) {
	.home {
		padding: 5% 0;
	}
}
