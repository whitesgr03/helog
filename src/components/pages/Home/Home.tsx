// Packages;
import { useState } from 'react';
import { Link } from 'react-router-dom';
import { useInfiniteQuery } from '@tanstack/react-query';

// Styles
import styles from './Home.module.css';
import imageStyles from '../../../styles/image.module.css';
import buttonStyles from '../../../styles/button.module.css';

// Components
import { Posts } from '../Post/Posts';
import { Loading } from '../../utils/Loading';

// Variables
import url from '../../../assets/hero.jpg';

// Utils
import { infiniteQueryPostsOption } from '../../../utils/queryOptions';

// Context
import { useAppDataAPI } from '../App/AppContext';

export const Home = () => {
	const { onAlert } = useAppDataAPI();
	const [isManuallyRefetch, setIsManuallyRefetch] = useState(false);

	const { isPending, isError, data, refetch } = useInfiniteQuery({
		...infiniteQueryPostsOption,
		meta: {
			errorAlert: () => {
				isManuallyRefetch &&
					onAlert([
						{
							message:
								'Loading the posts has some errors occur, please try again later.',
							error: true,
							delay: 4000,
						},
					]);
				setIsManuallyRefetch(false);
			},
		},
	});

	const posts = data?.pages[0].data.posts.slice(0, 4);

	const handleManuallyRefetch = () => {
		refetch();
		setIsManuallyRefetch(true);
	};

	return (
		<div className={styles.home}>
			<div className={styles.container}>
				<div className={styles.wrap}>
					<div className={imageStyles.content}>
						<img src={url} />
					</div>
				</div>
				<div className={styles.content}>
					<h2 className={styles.title}>
						Welcome to HeLog: A Daily Stories For Reader
					</h2>
					<p>
						Ever felt like your daily experiences are worth sharing? Those
						little moments that make you smile, the challenges that shaped you,
						or the random thoughts that keep you up at night? Well, you're not
						alone, and that's exactly why HeLog exists. Think of HeLog as your
						digital diary that connects with others. It's where your everyday
						experiences become part of a larger tapestry of human stories. No
						filters, no pretense - just real people sharing real moments. Let us
						make your story part of the world's story.
					</p>
					<Link to={'../posts'} className={styles.link}>
						All Posts
						<span
							className={`${imageStyles.icon} ${styles.doubleArrowRight}`}
						/>
					</Link>
				</div>
			</div>
			<div className={styles['latest-posts']}>
				<h2>Latest Posts</h2>
				{isError && !posts ? (
					<button
						className={`${buttonStyles.content} ${buttonStyles.more}`}
						onClick={handleManuallyRefetch}
					>
						Click here to load posts
					</button>
				) : isPending ? (
					<Loading text={'Loading posts ...'} />
				) : (
					<Posts posts={posts} />
				)}
			</div>
		</div>
	);
};
