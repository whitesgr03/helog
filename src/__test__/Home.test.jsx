import { vi, describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

import Home from "../components/Home";

vi.mock("../components/Posts");

describe("Home component", () => {
	it("should render correctly", () => {
		const { asFragment } = render(<Home />, { wrapper: BrowserRouter });

		const actual = asFragment();

		expect(actual).toMatchSnapshot();
	});
});
