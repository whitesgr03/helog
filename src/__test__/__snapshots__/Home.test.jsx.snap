// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home component > should render correctly 1`] = `
<DocumentFragment>
  <div
    class="_home_b140de"
  >
    <h2
      class="_title_b140de"
    >
      A Daily Story for Reader.
    </h2>
    <p>
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Odio, veniam voluptatem veritatis maxime iste rerum quo blanditiis cum ea nulla officiis neque molestiae voluptate qui adipisci velit magnam consequatur. Quo. Quia voluptas consequuntur assumenda ut totam quaerat explicabo eum nemo? Eveniet iste neque eligendi eius provident doloremque magnam maiores culpa repudiandae ipsam. Repudiandae illum facilis, omnis qui expedita debitis ea.
    </p>
    <a
      class="_link_b140de"
      href="/"
    >
      Latest Posts
      <svg
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m5.36 19l5.763-5.763a1.738 1.738 0 0 0 0-2.474L5.36 5m7 14l5.763-5.763a1.738 1.738 0 0 0 0-2.474L12.36 5"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
    </a>
  </div>
</DocumentFragment>
`;
